From 38193df20574ca82f05a3543b1b9fffcc98c474b Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Thu, 18 May 2023 14:57:59 +0900
Subject: [PATCH 20/20] HACK: Fix ARM_TIMER interurpts

This patch contains two fixes:
- Fix issue tpl_time_counter increment doesn't work.
- Fix Timer interrupt period

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 .../code/cortex-a/armv8/spider/counter_call.goilTemplate       | 2 +-
 machines/cortex-a/armv8/spider/tpl_machine_spider.c            | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/goil/templates/code/cortex-a/armv8/spider/counter_call.goilTemplate b/goil/templates/code/cortex-a/armv8/spider/counter_call.goilTemplate
index dd2aed34..d25e5ed1 100755
--- a/goil/templates/code/cortex-a/armv8/spider/counter_call.goilTemplate
+++ b/goil/templates/code/cortex-a/armv8/spider/counter_call.goilTemplate
@@ -28,7 +28,7 @@ FUNC(void, OS_CODE) tpl_tick_% !interrupt::NAME %()
     do
       %  tpl_counter_tick(&% !cpt_fct %_counter_desc);
 %
-  if interrupt::NAME == "SystemCounter" then
+  if interrupt::NAME == "ARM_TIMER" then
     %   tpl_time_counter++;
 %
   end if
diff --git a/machines/cortex-a/armv8/spider/tpl_machine_spider.c b/machines/cortex-a/armv8/spider/tpl_machine_spider.c
index 880e94a6..6840edcb 100644
--- a/machines/cortex-a/armv8/spider/tpl_machine_spider.c
+++ b/machines/cortex-a/armv8/spider/tpl_machine_spider.c
@@ -34,7 +34,8 @@
 #define CNTV_CTL_ENABLE   1
 #define CNTV_CTL_IMASK    2
 
-#define NB_TICK	((uint64)12500000 / (uint64)100) // systick 100 times per seconds (every 10ms)
+#define CPU_CLOCK (1060*1000*1000) // 1060 MHz
+#define NB_TICK	((uint64)(CPU_CLOCK/64) / (uint64)100) // systick 100 times per seconds (every 10ms)
 
 #define GICR_ISENABLER0     (0xf0110100)
 #define GICR_ICPENDR0       (0xf0110280)
-- 
2.34.1

